### PR TITLE
[ARCTIC-737][AMS] Capitalize the `authMethod` parameter.

### LINKED ISSUE #737: [Bug]: Not correctly equals `authMethod`
### What happened?

Create a catalog with kerberos authentication and execute sql in terminal.

```
java.lang.IllegalArgumentException: Error auth method:kerberos
	at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:217) ~[iceberg-bundled-guava-0.13.2.jar:?]
	at com.netease.arctic.table.TableMetaStore.<init>(TableMetaStore.java:121) ~[classes/:?]
	at com.netease.arctic.table.TableMetaStore.<init>(TableMetaStore.java:65) ~[classes/:?]
	at com.netease.arctic.table.TableMetaStore$Builder.build(TableMetaStore.java:635) ~[classes/:?]
	at com.netease.arctic.ams.server.terminal.TerminalManager.getCatalogTableMetaStore(TerminalManager.java:244) ~[classes/:?]
	at com.netease.arctic.ams.server.terminal.TerminalManager.executeScript(TerminalManager.java:92) ~[classes/:?]
	at com.netease.arctic.ams.server.controller.TerminalController.executeScript(TerminalController.java:70) ~[classes/:?]
```

### Affects Versions

0.4.0-SNAPSHOT

### What engines are you seeing the problem on?

AMS

### How to reproduce

Create a catalog with kerberos authentication and execute sql in terminal

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Fix #737 

## Brief change log

Capitalize the `authMethod` parameter.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/table/TableMetaStore.java [hunk 1]
```diff
@@ -542,7 +542,7 @@ public Builder withAuth(
         String authMethod, String hadoopUsername, byte[] krbKeyTabBytes, byte[] krbConfBytes,
         String krbPrincipal) {
       this.disableAuth = false;
-      this.authMethod = authMethod;
+      this.authMethod = authMethod == null ? null : authMethod.toUpperCase();
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/table/TableMetaStore.java:545]: I refer to add a statment after line 622 like this :
`authMethod  = authMethod.toUpperCase()`

And change line 609 to `String authMethod = properties.get(AUTH_METHOD).toUpperCase()`

How do you think?

**Review by zhoujinsong — COMMENTED** [review]
I left some comments. Please take another look.

File: core/src/main/java/com/netease/arctic/table/TableMetaStore.java (hunk 1 above)
Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/table/TableMetaStore.java:545]: The reason for this suggestion is that I hope to keep `Builder` the ability to accpet upper/lower case parameters and the build method is responsible for converting the case of the parameter to the right one.

Comment (zhongqishang) [reply core/src/main/java/com/netease/arctic/table/TableMetaStore.java:545]: The method `withKrbAuth` has the constant `AUTH_METHOD_SIMPLE/AUTH_METHOD_KERBEROS` assigned to it, so I think it's better not to have the value `kerberos` in `Builder`

> I hope to keep `Builder` the ability to accpet upper/lower case parameters

`withAuth` keep the ability to accpet upper/lower case parameters

Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/table/TableMetaStore.java:545]: Make sense to me, but you may still need to change the `Builder#readProperties` method to make it accept lowercase properties value.

Comment (zhongqishang) [reply core/src/main/java/com/netease/arctic/table/TableMetaStore.java:545]: done.

**Review by zhoujinsong — APPROVED** [review]
LGTM. Thanks a lot for your contribution!

**Pull request merged**